homeassistant/util/logging.py

```diff
@@ -164,3 +164,45 @@ def wrapper(*args: Any) -> None:
                 log_exception(*args)
         wrapper_func = wrapper
     return wrapper_func
+
+
+def catch_log_coro_exception(
+        target: Coroutine[Any, Any, Any],
+        format_err: Callable[..., Any],
+        *args: Any) -> Coroutine[Any, Any, Any]:
+    """Decorate a coroutine to catch and log exceptions."""
+    def log_exception(*args: Any) -> None:
+        module_name = inspect.getmodule(inspect.trace()[1][0]).__name__
+        # Do not print the wrapper in the traceback
+        frames = len(inspect.trace()) - 1
+        exc_msg = traceback.format_exc(-frames)
+        friendly_msg = format_err(*args)
+        logging.getLogger(module_name).error('%s\n%s', friendly_msg, exc_msg)
+
+    async def coro_wrapper(*args: Any) -> Any:
+        """Catch and log exception."""
+        try:
+            return await target
+        except Exception:  # pylint: disable=broad-except
+            log_exception(*args)
+            return None
+    return coro_wrapper()
+
+
+def async_create_catching_coro(
+        target: Coroutine) -> Coroutine:
+    """Wrap a coroutine to catch and log exceptions.
+
+    The exception will be logged together with a stacktrace of where the
+    coroutine was wrapped.
+
+    target: target coroutine.
+    """
+    trace = traceback.extract_stack()
```

Comment:
Not used?

Reply:
Used here:
https://github.com/home-assistant/home-assistant/blob/87f01b078c41917422f1ce51400edd3da02acdfc/homeassistant/util/logging.py#L206